seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceSplitEnumerator.java

```diff
@@ -369,7 +365,6 @@ public Map<TopicPartition, Long> listConsumerGroupOffsets(Collection<TopicPartit
 
     private void discoverySplits() throws ExecutionException, InterruptedException {
         fetchPendingPartitionSplit();
-        assignSplit();
```

Comment:
I think there might be a problem here. If all readers are registered and a new split be discover, It can not be assign.

Reply:
> I think there might be a problem here. If all readers are registered and a new split be discover, It can not be assign.

Got it

Reply:
Here @ic4y